src/middlewares/userMiddlewares.js

```diff
@@ -25,4 +25,25 @@ export default class UserMiddlewares {
       return resError(res, 500, error.message);
     }
   }
+
+  /**
+ * @name checkUserExistBeforeLogin
+ * @description Checks if a user exists in the database
+ * @param {object} req The request object
+ * @param {object} res The response object
+ * @param {object} next The response object
+ * @returns {object} The API response or next()
+ */
+  static async checkUserExistBeforeLogin(req, res, next) {
+    try {
+      const { email } = req.body;
+      const data = await userServices.getUserByEmail(email);
+      if (data) {
```

Comment:
You can assign the user object a value in the req before returning next(). Example `req.user = data`

Reply:
done so